### ISSUE TITLE
abbreviation expansion from dictionary changes the original text: adds fullstop

### ISSUE THREAD

**Issue by timobaumann** [marytts/marytts#757]
Tokenization+Normalization turns "Der Red Ball" into:
```xml
<?xml version="1.0" encoding="UTF-8"?><maryxml xmlns="http://mary.dfki.de/2002/MaryXML" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="0.5" xml:lang="de">
<p>
<s>
<t>
Der
</t>
<mtu accent="last" orig="Red.">
<t ending="noun" sounds_like="Redaktion">
Red.
</t>
</mtu>
<t>
Ball
</t>
<t>
Express
</t>
</s>
</p>
</maryxml>
```

Now, the point is not that the normalization is wrong, what we're concerned about is the added fullstop in the text of the document (it breaks other parts of our processing pipeline). 

Would you be opposed to a pull request that would get rid of the full stop after Red? At least for the orig-attribute? We're happy to implement the fix if you agree that this is in fact a bug.

**Comment by seblemaguer** [comment]
Hello,

I do agree it is a bug. I am currently developping the 6.X branch and the current master will be litterally changed. However, I didn't reach this abbreviation part so yes if you want to develop the fix that would be great. I will adapt it after to the next marytts version :)

**Comment by akoehn** [comment]
I wrote a small patch to fix the results of incorrect rules (not extensively tested):
[restore-orig-token.txt](https://github.com/marytts/marytts/files/1028907/restore-orig-token.txt)

Of course the rules shouldn't change the original, but there seem to be quite a few of them breaking our stuff.

See also [our bug tracking such breakages](https://bitbucket.org/natsuhh/swc/issues/51/)

**Comment by akoehn** [comment]
Maybe the patch could be expanded to throw a warning that the expansion rule should be fixed.  That way we would have correct expansions and there is still an incentive to fix the rules.

**Comment by timobaumann** [comment]
or, beyond a warning, put an assertion that getAttribute("orig")/getTextContent() should already be identical to origText. The fix looks good to me, what do you say, @seblemaguer?

**Comment by akoehn** [comment]
yes, I ran gradle test, but not gradle check, which indeed yields some errors with nested mtus. I'll investigate.

**Comment by timobaumann** [comment]
I wrote a new fix which specifically targets the problem of dot-insertion at the end of abbreviations (as in the example on the top). Please have a look at pull request #765.

**Comment by timobaumann** [comment]
we've since found a few more bugs in orig-incompatibility during normalization. See https://bitbucket.org/natsuhh/marytts/issues?status=new&status=open . 

We'd appreciate your input on whether you're willing to accept our changes (new test cases to previously untested code and fixes to keep the orig in line with the original text) and pushing that to a new jar on maven?

**Comment by psibre** [comment]
Sorry for not getting back to your sooner! The main issue is that these changes appear to deal only with the MaryXML processing, which is essentially deprecated in the master branch. If you could edit your PRs to go into the 5.x branch, we would be happy to merge it.
